pkg/rancher-desktop/config/__tests__/commandLineOptions.spec.ts

```diff
@@ -0,0 +1,390 @@
+
+        if (platform !== '*' &&
+            ((platform === '!win32' && os.platform() === 'win32') ||
+             (platform !== '!win32' && platform !== os.platform()))) {
+          expect(() => {
+            updateFromCommandLine(prefs, lockedSettings, [newOption]);
+          }).toThrow(`Changing field ${ accessor } via the API isn't supported`);
+        } else {
+          const newPrefs = updateFromCommandLine(prefs, lockedSettings, [newOption]);
+
+          expect(_.get(newPrefs, accessor)).toEqual(newValue);
+          _.set(newPrefs, accessor, oldValue);
+          expect(newPrefs).toEqual(origPrefs);
+        }
+      }
+    }
+  });
+
+  test('nothing after an = should set target to empty string', () => {
+    const newPrefs = updateFromCommandLine(prefs, lockedSettings, ['--images.namespace=']);
+
+    expect(origPrefs.images.namespace).not.toBe('');
+    expect(newPrefs.images.namespace).toBe('');
+    newPrefs.images.namespace = origPrefs.images.namespace;
+    expect(newPrefs).toEqual(origPrefs);
+  });
+
+  test('should change several values (and no others)', () => {
+    const newPrefs = updateFromCommandLine(prefs, lockedSettings, [
+      '--kubernetes.options.traefik=false',
+      '--application.telemetry.enabled=false',
+      '--portForwarding.includeKubernetesServices=true',
+      '--containerEngine.name=containerd',
+      '--kubernetes.port', '6444',
+    ]);
+
+    expect(newPrefs.kubernetes.options.traefik).toBeFalsy();
+    expect(newPrefs.application.telemetry.enabled).toBeFalsy();
+    expect(newPrefs.portForwarding.includeKubernetesServices).toBeTruthy();
+    expect(newPrefs.containerEngine.name).toBe('containerd');
+    expect(newPrefs.kubernetes.port).toBe(6444);
+
+    newPrefs.kubernetes.options.traefik = true;
+    newPrefs.application.telemetry.enabled = true;
+    newPrefs.portForwarding.includeKubernetesServices = false;
+    newPrefs.containerEngine.name = settings.ContainerEngine.MOBY;
+    newPrefs.kubernetes.port = 6443;
+    expect(newPrefs).toEqual(origPrefs);
+  });
+
+  test('should ignore non-option arguments', () => {
+    const arg = 'doesnt.start.with.dash.dash=some-value';
+    const newPrefs = updateFromCommandLine(prefs, lockedSettings, [arg]);
+
+    expect(newPrefs).toEqual(origPrefs);
+  });
+
+  test('should ignore an unrecognized option', () => {
+    const arg = '--kubernetes.zipperhead';
+    const newPrefs = updateFromCommandLine(prefs, lockedSettings, [arg]);
+
+    expect(newPrefs).toEqual(origPrefs);
+  });
+
+  test('should ignore leading options and arguments', () => {
+    const args = ['--kubernetes.zipperhead', '--another.unknown.option', 'its.argument', '--dont.know.what.this.is.either'];
+    const newPrefs = updateFromCommandLine(prefs, lockedSettings, args);
+
+    expect(TransientSettings.value.noModalDialogs).toEqual(false);
+    expect(newPrefs).toEqual(origPrefs);
+  });
+
+  test('should complain about an unrecognized ignore after a recognized one', () => {
```

Comment:
I don't think we have a test for recognized settings after unrecognized.: `['--ignore.this.one', '--kubernetes.enabled']`

Reply:
The test description was incorrect -- should be `option` not `ignore`.